integration-tests/src/integrationTest/java/pl/allegro/tech/hermes/integrationtests/MetricsTest.java

```diff
@@ -0,0 +1,357 @@
+                        )
+                        .withValue(1.0)
+                );
+    }
+
+    @Test
+    public void shouldReportHttpSuccessCodeMetrics() {
+        // given
+        TestSubscriber subscriber = subscribers.createSubscriber();
+        Topic topic = hermes.initHelper().createTopic(topicWithRandomName().build());
+        Subscription subscription = hermes.initHelper().createSubscription(
+                subscription(topic, "subscription")
+                        .withEndpoint(subscriber.getEndpoint())
+                        .withSubscriptionPolicy(
+                                subscriptionPolicy()
+                                        .applyDefaults()
+                                        .withMessageTtl(0)
+                                        .build()
+                        )
+                        .build()
+        );
+        TestMessage message = TestMessage.simple();
+
+        // when
+        hermes.api().publishUntilSuccess(topic.getQualifiedName(), message.body());
+
+        // then
+        subscriber.waitUntilReceived(message.body());
+        hermes.api().getConsumersMetrics()
+                .expectStatus()
+                .isOk()
+                .expectBody(String.class)
+                .value((body) -> assertThatMetrics(body)
+                        .contains("hermes_consumers_subscription_http_status_codes_total")
+                        .withLabels(
+                                "group", topic.getName().getGroupName(),
+                                "status_code", "200",
+                                "subscription", subscription.getName(),
+                                "topic", topic.getName().getName()
+                        )
+                        .withValue(1.0)
+                );
+    }
+
+    @Test
+    public void shouldReportMetricForFilteredSubscription() {
+        // given
+        TestSubscriber subscriber = subscribers.createSubscriber();
+        Topic topic = hermes.initHelper().createTopic(topicWithRandomName().build());
+        Subscription subscription = hermes.initHelper().createSubscription(
+                subscription(topic, "subscription")
+                        .withEndpoint(subscriber.getEndpoint())
+                        .withSubscriptionPolicy(
+                                subscriptionPolicy()
+                                        .applyDefaults()
+                                        .withMessageTtl(0)
+                                        .build()
+                        )
+                        .withFilter(filterMatchingHeaderByPattern("Trace-Id", "^vte.*"))
+                        .build()
+        );
+        TestMessage unfiltered = TestMessage.of("msg", "unfiltered");
+        TestMessage filtered = TestMessage.of("msg", "filtered");
+
+        // when
+        hermes.api().publishWithHeaders(topic.getQualifiedName(), unfiltered.body(), header("Trace-Id", "vte12"));
```

Comment:
perhaps these 2 should also be published until success?

Reply:
done